src/events/mod.rs

```diff
@@ -258,7 +258,7 @@ impl<'a> BytesStart<'a> {
         &'a self,
         attr_name: N,
     ) -> Result<Option<Attribute<'a>>> {
-        for a in self.attributes() {
+        for a in self.attributes().with_checks(false) {
```

Comment:
IMO, if you're asking for the first occurrence of one particular attribute, you probably don't care much about about getting an error back on finding some duplicate attribute.  It's almost twice as fast without the check.